File: mapper-annotation/src/main/java/pl/kpob/mapper_annotation/AutoMapping.java
package pl.kpob.mapper_annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by kpob on 16.12.2017.
 */

@Retention(RetentionPolicy.SOURCE)
@Target(ElementType.TYPE)
public @interface AutoMapping {
    boolean generateDomainModel() default true;
    boolean generateFirebaseModel() default true;
    boolean generateRepository() default true;
    boolean generateContract() default true;

}

File: mapper-annotation/src/main/java/pl/kpob/mapper_annotation/AutoMapping.kt
package pl.kpob.mapper_annotation


/**
 * Created by kpob on 16.12.2017.
 */

@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class AutoMapping(
        val generateDomainModel: Boolean = true,
        val generateFirebaseModel: Boolean = true,
        val generateRepository: Boolean = true,
        val generateContract: Boolean = true)
